src/qibo/models/qcnn.py

```diff
@@ -0,0 +1,307 @@
+        """
+        bias = np.array(theta[0 : self.measured_qubits])
+        angles = theta[self.measured_qubits :]
+
+        self.set_circuit_params(angles)
+        return self._circuit
+
+    def Predictions(self, circuit, theta, init_state, nshots=10000):
+        """
+        Args:
+            theta: list or numpy.array with the biases to be used in the circuit
+            init_state: numpy.array with the quantum state to be classified
+            nshots: int number of runs of the circuit during the sampling process (default=10000)
+        Returns:
+            numpy.array() with predictions for each qubit, for the initial state
+        """
+        bias = np.array(theta[0 : self.measured_qubits])
+        circuit_exec = circuit(init_state, nshots)
+        result = circuit_exec.frequencies(binary=False)
+        prediction = np.zeros(self.measured_qubits)
+
+        for qubit in range(self.measured_qubits):
+            for clase in range(self.nclasses):
+                binary = bin(clase)[2:].zfill(self.measured_qubits)
+                prediction[qubit] += result[clase] * (1 - 2 * int(binary[-qubit - 1]))
+
+        return prediction / nshots + bias
+
+    def square_loss(self, labels, predictions):
+        """
+        Args:
+            labels: list or numpy.array with the qubit labels of the quantum states to be classified
+            predictions: list or numpy.array with the qubit predictions for the quantum states to be classified
+        Returns:
+            numpy.float32 with the value of the square-loss function
+        """
+        loss = 0
+        for l, p in zip(labels, predictions):
+            for qubit in range(self.measured_qubits):
+                loss += (l[qubit] - p[qubit]) ** 2
+
+        return loss / len(labels)
+
+    def Cost_function(self, theta, data=None, labels=None, nshots=10000):
+        """
+        Args:
+            theta: list or numpy.array with the biases and the angles to be used in the circuit
+            nlayers: int number of layers of the varitional circuit ansatz
+            data: numpy.array data[page][word]  (this is an array of kets)
+            labels: list or numpy.array with the labels of the quantum states to be classified
+            nshots: int number of runs of the circuit during the sampling process (default=10000)
+        Returns:
+            numpy.float32 with the value of the square-loss function
+        """
+        circ = self.Classifier_circuit(theta)
+
+        Bias = np.array(theta[0 : self.measured_qubits])
+        predictions = np.zeros(shape=(len(data), self.measured_qubits))
+
+        for i, text in enumerate(data):
+            predictions[i] = self.Predictions(circ, Bias, text, nshots)
+
+        s = self.square_loss(labels, predictions)
+
+        return s
+
+    def minimize(
+        self, init_theta, data=None, labels=None, nshots=10000, method="Powell"
+    ):
+        """
+        Args:
```

Comment:
I think these docstring are wrong (e.g. `RY` is not passed here)! Can you have a look please?

Reply:
Thanks for the catch! Have corrected the docstrings.